### PR TITLE
8282548: Create a regression test for JDK-4330998

### PR DESCRIPTION
**Pull request by manukumarvs**
Create a regression test for [JDK-4330998](https://bugs.openjdk.java.net/browse/JDK-4330998)

Issue tested:
When calling JEditorPane.setText(null), the following exception is thrown:

java.lang.NullPointerException:
        at java.io.StringReader.<init>(StringReader.java:38)

Testing:
Java 1.4.0 -> Test Failed.
$ ./j2sdk1.4.0/bin/java bug4330998Win
java.lang.RuntimeException: Test failed, caught Exception java.lang.NullPointerException when calling text.setText(null)
        at bug4330998Win.runTest(bug4330998Win.java:45)
        at bug4330998Win.main(bug4330998Win.java:51)
Exception in thread "main"

Java 1.4.1 -> Test Passed.
$ ./j2sdk1.4.1/bin/java bug4330998Win
Test passed

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Issue
 * [JDK-8282548](https://bugs.openjdk.java.net/browse/JDK-8282548): Create a regression test for JDK-4330998

### Reviewers
 * [Alexey Ivanov](https://openjdk.java.net/census#aivanov) (@aivanov-jdk - **Reviewer**) ⚠️ Review applies to 03d7475a31c1b6dc171bfc5b2a617d1b158aaedf

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.java.net/jdk pull/7740/head:pull/7740` \
`$ git checkout pull/7740`

Update a local copy of the PR: \
`$ git checkout pull/7740` \
`$ git pull https://git.openjdk.java.net/jdk pull/7740/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 7740`

View PR using the GUI difftool: \
`$ git pr show -t 7740`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.java.net/jdk/pull/7740.diff">https://git.openjdk.java.net/jdk/pull/7740.diff</a>

</details>

### CODE REVIEW

**Comment by manukumarvs** [comment]
@aivanov-jdk Can you please review this.

**Comment by manukumarvs** [comment]
Any volunteers?

File: test/jdk/javax/swing/JEditorPane/4330998/JEditorPaneSetTextNullTest.java [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+  * @run main JEditorPaneSetTextNullTest
+  */
+ public class JEditorPaneSetTextNullTest {
+     private static JEditorPane editorPane;
+
+     public static void main(String[] s) throws Exception {
+         SwingUtilities.invokeAndWait(() -> createUI());
+
+         try {
+             SwingUtilities.invokeAndWait(() -> editorPane.setText(null));
```
Comment (aivanov-jdk) [inline test/jdk/javax/swing/JEditorPane/4330998/JEditorPaneSetTextNullTest.java:41]: I believe the test could be simplified to
```suggestion
             SwingUtilities.invokeAndWait(() -> new JEditorPane().setText(null));
```
And `createUI()` can be removed.

**Review by aivanov-jdk — CHANGES_REQUESTED** [review]

File: test/jdk/javax/swing/JEditorPane/4330998/JEditorPaneSetTextNullTest.java (hunk 1 above)
Comment (aivanov-jdk) [reply test/jdk/javax/swing/JEditorPane/4330998/JEditorPaneSetTextNullTest.java:41]: If you like to make sure the test expects `NullPointerException` only, you can use try-catch inside the lambda expression and propagate the exception thrown from `invokeAndWait`. Yet it's not required: we don't expect any exceptions thrown. Therefore try-catch can be dropped too.

Comment (aivanov-jdk) [reply test/jdk/javax/swing/JEditorPane/4330998/JEditorPaneSetTextNullTest.java]: > But if we want to throw the RuntimeException with a failure message, I think we need to catch it, right?

Yes, if you want to provide a more specific error message.

**Pull request closed without merging**